storage/innobase/log/log0recv.cc

```diff
@@ -1409,13 +1414,15 @@ static dberr_t recv_log_recover_10_4()
 	const lsn_t	lsn = log_sys.log.get_lsn();
 	const lsn_t	source_offset =	log_sys.log.calc_lsn_offset_old(lsn);
 	byte*		buf = log_sys.buf;
+	dberr_t		err;
 
 	if (!redo_file_sizes_are_correct()) {
 		return DB_CORRUPTION;
 	}
 
-	recv_sys.read(source_offset & ~lsn_t(OS_FILE_LOG_BLOCK_SIZE - 1),
-		      {buf, OS_FILE_LOG_BLOCK_SIZE});
+	if ((err= recv_sys.read(source_offset & ~lsn_t(OS_FILE_LOG_BLOCK_SIZE - 1),
+		      {buf, OS_FILE_LOG_BLOCK_SIZE})))
+		return err;
```

Comment:
I’d declare the variable `err` just inside the `if` statement, in C++98 style.

Reply:
The line becomes too long. To avoid that and to be somewhat in line with the old-style formatting, I would suggest the following:
```c++
	if (dberr_t err =
            recv_sys.read(source_offset & ~lsn_t(OS_FILE_LOG_BLOCK_SIZE - 1),
			  {buf, OS_FILE_LOG_BLOCK_SIZE})) {
		return err;
	}
```